auth/auth.go

```diff
@@ -465,17 +465,23 @@ func (auth *Authenticator) InvalidateChannels(name string, isUser bool, scope st
 
 	base.InfofCtx(auth.LogCtx, base.KeyAccess, "Invalidate access of %q", base.UD(name))
 
-	subdocPath := "channel_inval_seq"
-	if scope != base.DefaultScope || collection != base.DefaultCollection {
-		subdocPath = "collection_access." + scope + "." + collection + "." + subdocPath
-	}
+	// Attempt to use subdoc if we're only modifying a single collection
+	if len(collections) == 1 {
```

Comment:
This is more of an optimization than might be necessary, but can we change the API of SubdocInsert to be map[string]any so we can do multiple subdoc operations at once?

Later we are going to run through the logic of updating the document, though not the write operation.

Reply:
I considered adding that functionality in this PR, but decided against it due to the increase in scope/risk.  Since the more common usage of InvalidateChannels (at write time) is always single collection, I thought it was reasonable to omit.